### PR TITLE
Section on current challenges

### PR DESCRIPTION
**Pull request by nicocartesi**
Section describing current challenges for the Rollups Framework development. This is a continuation of the internal PR https://github.com/cartesi-corp/docs/pull/241

### CODE REVIEW

File: cartesi-rollups/challenges/overview.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+## Disputes
+
+### Challenge
+
+The **1.0** release lacks support for dispute resolution.
+
+### Solution strategy
+
+From its inception, the Cartesi framework has the potential to integrate dispute resolution features. Cartesi's community-driven iterations have leveraged dispute resolution for [Compute](/compute/overview). Prioritizing other features and improvements, like the Microarchitecture enhancement and recent [research on NxN dispute resolution](https://arxiv.org/abs/2212.12439), the community might delve deeper into this area.
```
Comment (gbarros) [inline cartesi-rollups/challenges/overview.md:34]: This seems to be the wrong arxiv link. I don't have the correct one tho.

**Comment by gbarros** [comment]
Reading further, I think we are all good with the rest of the text =) 👍

File: cartesi-rollups/challenges/overview.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+## Transaction scalability
+
+### Challenge
+
+In contrast to the majority of proposed Layer 2 rollup solutions, which primarily focus on enhancing transaction throughput and minimizing fees (essentially facilitating more transactions at a lower cost), the current **v1.0** release is dedicated to expanding the *computational capacity* (as opposed to *transaction scalability* alone) to allow the creation of decentralized applications that were simply not possible to build before.
+
+### Solution strategy
+
+One potential path to achieve transaction scalability is the introduction of an *Aggregator*, often associated with a Sequencer. This component might enhance the overall efficiency and capacity of the rollup solution. Moreover, an exciting area for exploration is deploying Cartesi on top of other Layer-2 solutions like Optimism and Arbitrum, potentially expanding the options for enhanced transaction scalability.
```
Comment (jjhbk) [inline cartesi-rollups/challenges/overview.md:24]: 'A potential path that can be considered '

File: cartesi-rollups/challenges/overview.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+## Composability
+
+### Challenge
+
+The current **v1.0** release lacks composability, which is the ability of multiple DApps to directly interact with each other, for example by having one application directly send inputs to another one without going through the base layer. This would enable seamless communication and data exchange between DApps. The most important use case for this feature would be to allow users to easily and cheaply transfer assets from one DApp to another. 
+
+### Solution strategy
+
+Post the **v1.0** mainnet release, a potential area of focus could be research into addressing composability, vital for enhancing DApp functionality. Meanwhile, interim solutions like leveraging liquidity providers for asset transfers might be considered, enhancing the user experience while juggling development priorities.
```
Comment (jjhbk) [inline cartesi-rollups/challenges/overview.md:58]: Is there any specific area to focus in the research which can be suggested here?

File: cartesi-rollups/challenges/overview.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+In particular, upgrading the back-end code of a DApp is an important concern that may arise due to the discovery of a bug in the application itself or the software stack that it utilizes. This requires the DApp's Cartesi Machine to be safely updated, for which there is currently no standard mechanism or guideline. As a reminder, individual DApps can of course always choose to simply not be upgradable at all.
+
+### Solution strategy
+
+Following the **v1.0** mainnet release, an interesting area for research could be to explore methods for supporting an upgrade procedure for DApps that are inclined to allow it. A key concern in this research domain is finding a way, or a set of best practices, to securely handle asset ownership continuity – for instance, how to transfer assets safely from an older version of an application to its newer incarnation. Given that an updated DApp would entail utilizing a different Cartesi Machine, the challenge is to devise a robust method to migrate the previous machine's state, inclusive of user assets, into this new framework.
```
Comment (jjhbk) [inline cartesi-rollups/challenges/overview.md:80]: 'for instance, how to maintain asset ownership of the DApp after a version upgrade'

Comment (jjhbk) [inline cartesi-rollups/challenges/overview.md:80]: 'Given that an updated DApp would entail utilizing a different Cartesi Machine, a possible challenge is to devise a robust method to migrate the previous machine's state, inclusive of user assets, into this new framework'

File: cartesi-rollups/challenges/overview.md (hunk 3 above)
Comment (nicocartesi) [reply cartesi-rollups/challenges/overview.md]: I have reworded this sentence in line with the one above

**Comment by Payalpatelmalhotra** [comment]
@nicocartesi The following needs to be reworded please, I am not quite understanding what we are saying here and if you review the sentence, the flow is broken:

Disputes > solution strategy

Prioritizing other features and improvements, like the Microarchitecture enhancement and recent [research on NxN dispute resolution](https://arxiv.org/abs/2212.12439), the community might delve deeper into this area.

File: cartesi-rollups/challenges/overview.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+Another feature for community consideration could be a mechanism assisting front-end clients in dividing large pieces of data into smaller chunks and reassembling them within the back-end. This approach would be similar to the [Logger](/compute/logger_drive/) feature used in Cartesi Compute.
+
+Additionally, potential future exploration might focus on integrating Rollups with Data Availability solutions like zkPorter or using a Data Availability Committee solution for quorum-based validator sets.
+
+## Composability
+
+### Challenge
+
+The current **v1.0** release lacks composability, which is the ability of multiple DApps to directly interact with each other, for example by having one application directly send inputs to another one without going through the base layer. This would enable seamless communication and data exchange between DApps. The most important use case for this feature would be to allow users to easily and cheaply transfer assets from one DApp to another.
```
Comment (felipeargento) [inline cartesi-rollups/challenges/overview.md:54]: I wouldn't say "lacks composability"! Cartesi Rollups do have composability in the form of vouchers, so it is as generic as possible. The problem is in the price (its an L1 tx) and the delay (has to wait for the challenge period). 

I'd say something like: "The current **v1.0** release doesnt offer great composability" or "The current release still faces a lot of composability hurdles". Something that doesnt imply the inexistence of composability

**Pull request merged**